src/mongocrypt.c

```diff
@@ -523,5 +523,112 @@ mongocrypt_setopt_crypto_hooks (mongocrypt_t *crypt,
    }
    crypt->crypto->sha_256 = sha_256;
 
+   return true;
+}
+
+bool
+mongocrypt_setopt_kms_providers (mongocrypt_t *crypt,
+                                 mongocrypt_binary_t *kms_providers)
+{
+   mongocrypt_status_t *status;
+   bson_t as_bson;
+   bson_iter_t iter;
+
+   if (!crypt) {
+      return false;
```

Comment:
The `status` is attached to `crypt`, so there's no way to return it to the caller. Passing `null` is API misuse. Developers could hit this when wrapping the bindings, but it would not happen in regular use of libmongocrypt.